youtube_dl/extractor/common.py

```diff
@@ -2167,7 +2167,14 @@ def _find_jwplayer_data(webpage):
             r'jwplayer\((?P<quote>[\'"])[^\'" ]+(?P=quote)\)\.setup\s*\((?P<options>[^)]+)\)',
             webpage)
         if mobj:
-            return mobj.group('options')
+            options = mobj.group('options').strip()
+            # We're looking for JSON
+            if options[0] != '{':
+                # setup() got a variable as an argument - resolve it
+                regex = r'%s\s*=\s*(?P<options>\{[^\}]\})' % options
+                mobj = re.search(regex, webpage)
+                if mobj:
```

Comment:
If there's no match here, options should still be returned as it might be really a valid JSON. For example:
```
jwplayer('foobar').setup(
    /* some comments */
    {
        'file': 'foobar.mp4'
    }
);
```